tls/s2n_client_key_exchange.c

```diff
@@ -32,7 +32,7 @@
 #include "utils/s2n_random.h"
 #include "utils/s2n_safety.h"
 
-#define get_client_hello_protocol_version(conn) (conn->client_hello_version == S2N_SSLv2 ? conn->client_protocol_version : conn->client_hello_version)
+#define get_client_hello_protocol_version(conn) (conn->client_hello.version == S2N_SSLv2 ? conn->client_protocol_version : conn->client_hello.version)
```

Comment:
I found this macro confusing. My understanding after thinking about it for a bit is as follows:

It's not possible to negotiate the SSLv2 so if the `client_hello` says we are using SSLv2 we need to check the `client_protocol_version` instead. 

But I'm confused why we don't just use client_protocol_version? Is it not in sync with client_hello.version at some moment?

Perhaps this comment is relevant:

```
/* client_hello.version is set when parsing the record header for
     * SSLv2 ClientHellos, not when parsing the ClientHello itself.
     * Therefore, client_hello.version is meaningful ONLY for SSLv2
     * before we parse the ClientHello.
     */
```